benchmarks/src/main/java/io/grpc/benchmarks/qps/Utils.java

```diff
@@ -0,0 +1,343 @@
+import io.grpc.transport.netty.NettyServerBuilder;
+import io.grpc.transport.okhttp.OkHttpChannelBuilder;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.ServerChannel;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslProvider;
+
+import org.HdrHistogram.Histogram;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+
+/**
+ * Utility methods to support benchmarking classes.
+ */
+final class Utils {
+  private static final String UNIX_DOMAIN_SOCKET_PREFIX = "unix://";
+
+  // The histogram can record values between 1 microsecond and 1 min.
+  static final long HISTOGRAM_MAX_VALUE = 60000000L;
+  // Value quantization will be no larger than 1/10^3 = 0.1%.
+  static final int HISTOGRAM_PRECISION = 3;
+
+  private Utils() {
+  }
+
+  static boolean parseBoolean(String value) {
+    return value.isEmpty() || Boolean.parseBoolean(value);
+  }
+
+  static SocketAddress parseSocketAddress(String value) {
+    if (value.startsWith(UNIX_DOMAIN_SOCKET_PREFIX)) {
+      // Unix Domain Socket address.
+      try {
+        // Create the underlying file for the Unix Domain Socket.
+        String filePath = value.substring(UNIX_DOMAIN_SOCKET_PREFIX.length());
+        File file = new File(filePath);
+        if (!file.isAbsolute()) {
+          throw new IllegalArgumentException("File path must be absolute: " + filePath);
+        }
+        if (file.createNewFile()) {
+          // If this application created the file, delete it when the application exits.
+          file.deleteOnExit();
+        }
+        // Create the SocketAddress referencing the file.
+        Class<?> addressClass = Class.forName("io.netty.channel.unix.DomainSocketAddress");
+        return (SocketAddress) addressClass.getDeclaredConstructor(File.class)
+            .newInstance(file);
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    } else {
+      // Standard TCP/IP address.
+      String[] parts = value.split(":");
+      String host = parts[0];
+      int port = Integer.parseInt(parts[1]);
+      return new InetSocketAddress(host, port);
+    }
+  }
+
+  static SimpleRequest newRequest(ClientConfiguration config) {
+    ByteString body = ByteString.copyFrom(new byte[config.clientPayload]);
+    Payload payload = Payload.newBuilder()
+        .setType(config.payloadType)
+        .setBody(body)
+        .build();
+
+    return SimpleRequest.newBuilder()
+        .setResponseType(config.payloadType)
+        .setResponseSize(config.serverPayload)
+        .setPayload(payload)
+        .build();
+  }
+
+  static ServerImpl newServer(ServerConfiguration config) throws IOException {
```

Comment:
It doesn't bother me too much, but what do we gain by putting server construction in a Util class? For client-side, it was a requirement to share code, and instead of inheritance a Util class was used. I don't see the benefit of doing that on server-side, since it was only done on client-side because we had to.